frontend/src/views/DashboardPage/components/MoveSecrets/DirectorySelector.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useEffect, useState } from "react";
+import { useRouter } from "next/router";
+import { twMerge } from "tailwind-merge";
+
+import { TSecretFolder } from "../../../../hooks/api/secretFolders/types";
+
+
+interface Folder {
+    name: string;
+    id: string;
+    folders?: Folder[];
+}
+
+interface DirectorySelectorProps {
+    directoryData: {
+        dir: TSecretFolder[];
+        folders: TSecretFolder[];
+    } | undefined;
+    onSelect: (selectedPath: string, folderId: string) => void;
+}
+
+const DirectorySelector: React.FC<DirectorySelectorProps> = ({ directoryData, onSelect }) => {
+    const [selectedPath, setSelectedPath] = useState("");
+    const [directoryTree, setDirectoryTree] = useState<TSecretFolder[]>([])
+    const [selectedFolder, setSelectedFolder] = useState<string>("")
+
+    const router = useRouter();
+    const { query } = router;
+    const queryFolderId = query.folderId;
+
+    useEffect(() => {
+        if ((directoryData?.dir && directoryData?.dir.length > 0)) {
+            const directoryDataCopy = [...directoryData.dir]
+            const $directoryTree: TSecretFolder[] = [];
+
+            let i = 0;
+
+            while (i < directoryDataCopy.length) {
+                const currentItem = directoryDataCopy[i];
+                const parentItem = directoryDataCopy[i - 1];
+
+                if (parentItem) {
+                    if (!parentItem.folders) {
+                        parentItem.folders = [];
+                    }
+                    if (currentItem.id !== queryFolderId) {
+                        if (!parentItem.folders.find(folder => folder.id === currentItem.id)) {
+                            parentItem.folders.push(currentItem);
+                        }
+                    }
+
+                } else if (currentItem.id !== queryFolderId) {
+                    $directoryTree.push(currentItem);
+                }
+
+                i += 1;
+            }
+            setDirectoryTree(() => $directoryTree)
+        }
+    }, [directoryData])
+
+    const handlePathSelection = (path: string, folderId: string) => {
+        setSelectedFolder(folderId)
+        setSelectedPath(path);
+        onSelect(path, folderId);
+    };
+
+
+    const renderRootDirectory = (folders: Folder[] | undefined, currentPath: string, depth = 0) => {
+        if (!folders) return null;
+
+        const indentation = depth * 20;
+
+        return (
+            <ul style={{ marginLeft: `${indentation}px` }}>
+                {folders.map((folder) => (
+                    <li key={Math.floor(Math.random() * 50000) + 1}>
+                        <button
+                            className={twMerge(`py-1 px-2 bg-mineshaft-500  hover:bg-mineshaft-600 mb-2 rounded-md ${selectedFolder === folder.id ? "bg-mineshaft-400 hover:bg-mineshaft-400" : ""}`)}
```

Comment:
Using `twMerge` u dont need to string concat like this, u can use comma seperated truthy and falsy to strip of class conditionally. Kindly refer twMerge documentation

Reply:
Addressed!

Reply:
@EBEN4REAL Could you use `folder.id` as the **key** in the `<li>` element instead of using `Math.random()`?

https://react.dev/learn/rendering-lists#rules-of-keys